cabal-install/Distribution/Client/DistDirLayout.hs

```diff
@@ -107,14 +111,20 @@ data DistDirLayout = DistDirLayout {
      }
 
 
+data StoreDirLayout = StoreDirLayout {
```

Comment:
What about storing all dynamic libraries in a single directory (as is needed on Mac OS X, because of RPATH length limit?)